packages/coding-standard/src/Fixer/LineLength/DocBlockLineLengthFixer.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Fixer\LineLength;
+
+use Nette\Utils\Strings;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use SplFileInfo;
+use Symplify\CodingStandard\Fixer\AbstractSymplifyFixer;
+use Symplify\RuleDocGenerator\Contract\ConfigurableRuleInterface;
+use Symplify\RuleDocGenerator\Contract\DocumentedRuleInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Symplify\CodingStandard\Tests\Fixer\LineLength\DocBlockLineLengthFixer\DocBlockLineLengthFixerTest
+ */
+final class DocBlockLineLengthFixer extends AbstractSymplifyFixer implements ConfigurableRuleInterface, ConfigurableFixerInterface, DocumentedRuleInterface
+{
+    /**
+     * @api
+     * @var string
+     */
+    public const LINE_LENGTH = 'line_length';
+
+    /**
+     * @var string
+     */
+    private const ERROR_MESSAGE = 'Docblock lenght should fit expected width';
+
+    /**
+     * @var int
+     */
+    private $lineLength = 120;
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(self::ERROR_MESSAGE, []);
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    public function fix(SplFileInfo $file, Tokens $tokens): void
+    {
+        // function arguments, function call parameters, lambda use()
+        for ($position = count($tokens) - 1; $position >= 0; --$position) {
+            /** @var Token $token */
+            $token = $tokens[$position];
+            if (! $token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            $docBlockLines = explode(PHP_EOL, $token->getContent());
+            foreach ($docBlockLines as $docBlockLine) {
+                if (Strings::length($docBlockLine) <= $this->lineLength) {
+                    continue;
+                }
+
+                $extraDocBlockLines = str_split($docBlockLine, $this->lineLength);
```

Comment:
@matthiasnoback This is very dummy first prototype. It's crappy :)

Any idea how could we split long docblock text differently?

Reply:
I think we need a proper word wrapper for the description block. E.g. find the last space starting from the maximum line length. There is probably some simple function somewhere that could be used here.

Reply:
It seems that's what this function does, so I don't understand what exactly you mean.
Could you share some code/package?

Reply:
As far as I could tell this would break a sentence in the middle of a word if it crosses the maximum line length, right? Or maybe I don't get it.

Reply:
What's relevant too is that the maximum line length applies to the whole line, not just the doc block line. It would be nice to show that a doc block at some indentation level doesn't cross the maximum line length either.

I'm thinking about a few other scenarios:

- The line can't be split by normal word-wrapping, e.g. if it contains a long URL. In that case the sentence should just be split at the last possible character (I'd say).
- In the case of a very long class name being used in the description, maybe it could be wrapped at more logical points, e.g. `\\` or `[A-Z]`.
- There should be a `skip` fixture that shows that tags aren't modified (e.g. `@param`), and that this is just about the description.

Reply:
@matthiasnoback `wordwrap` has a parameter to break long words (https://www.php.net/manual/de/function.wordwrap.php) if they exceed the specified width. Could probably be used for the "long URL" scenario. Using something like `preg_split("/\n/", wordwrap($docBlockLine, $this->lineLength, "\n", true))` should not break in the middle of a word but cut very long words (like URLs). Not sure how this could be extended to cases like class names where you'd not want to break at an arbitrary point though, probably more mature logic is needed for this.

Reply:
Could you share a working example on https://3v4l.org/?

Reply:
https://3v4l.org/NfCN1

This displays the problem as I understood it by @matthiasnoback description. The word "should" is broken in the middle when using str_split. I've also added the wordwrap example to highlight the difference.

Reply:
Indeed, `wordwrap` will do the trick for now I'm sure. Here's something else I was thinking of:

Before:
```
/*
 * This is a very long line. This is a very long line. There are more words here.
 * And even more words here.
 */ 
```

After (incorrect):
```
/*
 * This is a very long line. This is a very long line. There
 * are more words here.
 * And even more words here.
 */ 
```
After (correct):
```
/*
 * This is a very long line. This is a very long line. There
 * are more words here. And even more words here.
 */ 
```
I think we should remove existing `\n`s and wrap the result to a maximum line length.
Double `\n`s should be taken as a paragraph break so automatic wrapping should start again there.

Reply:
Now that I'm coming up with all these requirements, would it be appreciated if I continue your work here, @TomasVotruba ? Or is this something you're interested to work on?

Reply:
> Now that I'm coming up with all these requirements, would it be appreciated if I continue your work here

That would be the best,  as you have the idea :+1: go for it!

(I have dozens issues/PRs on Rector today :D )

Reply:
Cool, it's WIP but I think it'll work.